### ISSUE TITLE
Crashing on loading dylibs

### ISSUE THREAD

**Issue by wajsic** [kabiroberai/theos-jailed#74]
Hello! I am trying to inject a simple dylib but it looks like there is something wrong with patching

When attaching debugger right on launch I get the error "load commands exceed length of first segment"
Now doing some research it leads me to [this line](
https://github.com/PureDarwin/dyld/blob/56cd028e61d0d487e5b604eb6bd2c5d577b24241/dyld3/MachOFile.cpp#L610) in Apple's dyld3 module which looks quite self-explanatory. Now has anyone encountered similar issues? If so are there any workarounds or shall I go debug optool? 

Thanks

**Comment by kabiroberai** [comment]
Hi, it's possible that you're seeing this error due to the app binary not having enough space between its load commands and the rest of the __TEXT segment. Is it specific to a single app or are you seeing it across the board?

**Comment by wajsic** [comment]
I tried with 2 different apps and had the same issues with both. I created a clean app with just a simple dylib to test injection and got it straight away. I can upload IPA and a tweak, if you want to repro. But I did have success before with larger apps so could be related to really small binaries. I tried other tools that also use optool and all came out with same crash so I suspect it's optool's fault but all the repos are dead so I created issue here.

**Comment by kabiroberai** [comment]
Yeah it'd be great if you could send me a reproducible example, thanks!
